### PR TITLE
chore: bump github.com/prometheus/prometheus to v0.45.0

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: pkg/runutil/runutil.go [hunk 1]
```diff
@@ -68,31 +67,6 @@ func ExhaustCloseWithLogOnErr(logger log.Logger, r io.ReadCloser, format string,
...
-	_, copyErr := io.Copy(ioutil.Discard, r)
-
-	CloseWithErrCapture(err, r, format, a...)
-
-	// Prepend the io.Copy error.
-	merr := tsdberrors.MultiError{}
-	merr.Add(copyErr)
-	merr.Add(*err)
-
-	*err = merr.Err()
```
Comment (philipgough) [inline pkg/runutil/runutil.go:93]: were these unused?

File: cmd/rhelemeter-server/main_e2e_test.go [hunk 2]
```diff
@@ -16,13 +16,13 @@ import (
...
 
-	"github.com/go-kit/kit/log"
+	"github.com/go-kit/log"
 	"github.com/gogo/protobuf/proto"
 	"github.com/golang/snappy"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/expfmt"
 	"github.com/prometheus/prometheus/prompb"
-	"github.com/prometheus/prometheus/util/testutil"
+	"github.com/stretchr/testify/require"
```
Comment (saswatamcode) [inline cmd/rhelemeter-server/main_e2e_test.go:25]: Hmm can we keep the `testutil` notation here and replace with: https://github.com/efficientgo/core/tree/main/testutil? This is what we use throughout Thanos as well, so would make this repo much more consistent.

**Review by saswatamcode — APPROVED** [review]
Thanks! Generally LGTM! :)
Some comments.

File: pkg/runutil/runutil.go (hunk 1 above)
Comment (simonpasquier) [reply pkg/runutil/runutil.go:93]: correct :)

File: pkg/runutil/runutil.go
[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/runutil/runutil.go:1]: I'll keep the recommendation for a follow-up PR.

**Review by philipgough — APPROVED** [review]

**Pull request merged**